controllers/image_generator.go

```diff
@@ -133,6 +134,12 @@ func ImageCreate(c client.Client) (int, error) {
 	}
 
 	ig := GetImageGenerator()
+
+	if ig == nil {
```

Comment:
It would be good to understand how GetImageGenerator is returning a nil ig. As per the sequence I don't understand how this is happening.
ImageCrate -> InitializeImageGenerator -> newImageGenerator -> initialises ig

Reply:
Ok, I understand this now..  Probably some of the conditions in the newImageGenerator fails in your case

Reply:
Thanks Pradipta..

Reply:
What about extending `GetImageGenerator` to error when `ig` is `nil`? Then manage the error here. This could make it clearer what to expect for future interactions with this code.